compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -993,9 +997,21 @@ public void visit(BLangIntersectionTypeNode intersectionTypeNode, AnalyzerData d
 
     @Override
     public void visit(BLangTupleTypeNode tupleTypeNode, AnalyzerData data) {
-        List<BLangType> memberTypeNodes = tupleTypeNode.memberTypeNodes;
-        for (BLangType memType : memberTypeNodes) {
-            analyzeNode(memType, data);
+        List<BLangSimpleVariable> memberNodes = tupleTypeNode.members;
+        BType bType = tupleTypeNode.getBType();
+        data.env = bType.tag == TypeTags.NONE ? data.env :
+                SymbolEnv.createTypeEnv(tupleTypeNode, new Scope(bType.tsymbol), data.env);
```

Comment:
Why do we perform this check? Here, How does the type tag become "None"?

Reply:
if bType is `noType` we won't to avoid updating the environment. Should be changed to `bType == symTable.noType`.

Reply:
Can you provide an example for this? In my opinion, when it comes to the semantic analyzer, the type cannot be `notype`

Reply:
```
type T [B, C];

type B int;
```
if C is undefined bType of T would be `noType` here right?

Reply:
Currently, we set the type of T as `noType`. But, the type of T should be `[int, noType]`.